controllers/storagecluster/storagecluster_controller_test.go

```diff
@@ -192,6 +192,111 @@ var mockDeviceSets = []api.StorageDeviceSet{
 	},
 }
 
+func getMockStorageProfiles() *api.StorageProfileList {
+	const pgAutoscaleMode = "pg_autoscale_mode"
+	const pgNum = "pg_num"
+	const pgpNum = "pgp_num"
+	const namespace = ""
+	const kind = "StorageProfile"
+	const apiVersion = "ocs.openshift.io/v1"
+	spfast := &api.StorageProfile{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fast-performance",
+			Namespace: namespace,
+		},
+		Spec: api.StorageProfileSpec{
+			DeviceClass: "fast",
+			BlockPoolConfiguration: api.BlockPoolConfigurationSpec{
+				Parameters: map[string]string{
+					pgAutoscaleMode: "on",
+					pgNum:           "128",
+					pgpNum:          "128",
+				},
+			},
+			SharedFilesystemConfiguration: api.SharedFilesystemConfigurationSpec{
+				Parameters: map[string]string{
+					pgAutoscaleMode: "on",
+					pgNum:           "128",
+					pgpNum:          "128",
+				},
+			},
+		},
+	}
+	spmed := &api.StorageProfile{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "med-performance",
+			Namespace: namespace,
+		},
+		Spec: api.StorageProfileSpec{
+			DeviceClass: "med",
+			BlockPoolConfiguration: api.BlockPoolConfigurationSpec{
+				Parameters: map[string]string{
+					pgAutoscaleMode: "on",
+					pgNum:           "128",
+					pgpNum:          "128",
+				},
+			},
+			SharedFilesystemConfiguration: api.SharedFilesystemConfigurationSpec{
+				Parameters: map[string]string{
+					pgAutoscaleMode: "on",
+					pgNum:           "128",
+					pgpNum:          "128",
+				},
+			},
+		},
+	}
+	spslow := &api.StorageProfile{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "slow-performance",
+			Namespace: namespace,
+		},
+		Spec: api.StorageProfileSpec{
+			DeviceClass: "slow",
+			BlockPoolConfiguration: api.BlockPoolConfigurationSpec{
+				Parameters: map[string]string{
+					pgAutoscaleMode: "on",
+					pgNum:           "128",
+					pgpNum:          "128",
+				},
+			},
+			SharedFilesystemConfiguration: api.SharedFilesystemConfigurationSpec{
+				Parameters: map[string]string{
+					pgAutoscaleMode: "on",
+					pgNum:           "128",
+					pgpNum:          "128",
+				},
+			},
+		},
+	}
+	spblankdeviceclass := &api.StorageProfile{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "blank-performance",
```

Comment:
What does blank performance mean? a default or low performance?

Reply:
A `StorageProfile` is marked as `Rejected` in its phase (also referred to as invalid) when its device class field is blank.

Reply:
Resolving as per @rchikatw's manual confirmation